### PR TITLE
Fix #11395: Change page URL helper function to no longer alter passed in URL

### LINKED ISSUE #11395: Add ability for `page()` URL helper to accept URL names with trailing slash
Doing this will add consistency with the built-in Django `path()` method. Confusion around this has come up several times recently. The [function is defined here](https://github.com/mozilla/bedrock/blob/7690d95755f8387fd480d159a6eb320a2d1fac75/bedrock/mozorg/util.py#L24).

### PR DESCRIPTION
**Pull request by pmac**
The `page()` helper used to convert the URL `name` passed in to a regex and add a trailing slash. None of the uses of this function needed it to be a regex, and now that django has a more simple `path` method for URLs, this PR switches the `page` helper to just pass through the `name` argument directly to the `path` helper. This also updates all uses of the `page` helper and the docs.

### CODE REVIEW

File: docs/coding.rst [hunk 1]
```diff
@@ -170,30 +170,29 @@ in our `ESLint config <https://github.com/mozilla/bedrock/blob/master/.eslintrc.
...
 --------------------
 
-URL patterns should be as strict as possible. It should begin with a
-`^` and end with `/$` to make sure it only matches what you specifiy.
-It also forces a trailing slash. You should also give the URL a name
-so that other pages can reference it instead of hardcoding the URL.
-Example:
+URL patterns should be as strict as possible. It also forces a trailing slash.
+You should also give the URL a name so that other pages can reference it instead
+of hardcoding the URL. Example:
```
Comment (robhudson) [inline docs/coding.rst:175]: This reads a bit awkward to me. Is "It" referring to URL patterns? Also I don't think this matches the code any longer, the trailing slash is more explicitly defined in each URL now but optional still, it looks like.

Comment (pmac) [reply docs/coding.rst:175]: Should no longer be optional. I removed the optional addition of the URL as well.

File: bedrock/mozorg/util.py [hunk 2]
```diff
@@ -30,9 +30,7 @@ def page(name, tmpl, decorators=None, url_name=None, ftl_files=None, **kwargs):
     page's URL name will be "path.to.template". Set the `url_name` parameter
     to override this name.
 
-    @param name: The URL regex pattern.  If not empty, a trailing slash is
-        added automatically, so it shouldn't be included in the parameter
-        value.
+    @param name: The URL path. It is passed to `django.urls.path`.
```
Comment (robhudson) [inline bedrock/mozorg/util.py:33]: A test for a slash-less path would be good. Or even docs to say you can do something like `page("path/to/file.json", ...)`, if needed.

File: docs/coding.rst (hunk 1 above)
Comment (pmac) [reply docs/coding.rst]: Updated wording.

File: bedrock/mozorg/util.py (hunk 2 above)
Comment (pmac) [reply bedrock/mozorg/util.py:33]: Added test and a link to the Django docs in the docs.

**Review by robhudson — APPROVED** [review]
r+wc

**Pull request merged**